Sources/SwiftParser/Types.swift

```diff
@@ -337,8 +337,15 @@ extension Parser {
     )
   }
 
-  mutating func parseTypeNameWithGenerics(include flags: DeclNameOptions = []) -> (RawTokenSyntax, RawGenericArgumentClauseSyntax?) {
-    let (name, _) = self.parseDeclNameRef(flags)
+  mutating func parseTypeNameWithGenerics(allowKeywordAsName: Bool) -> (RawTokenSyntax, RawGenericArgumentClauseSyntax?) {
+    let name: RawTokenSyntax
+    if let identOrSelf = self.consume(if: .identifier, .keyword(.Self), .keyword(.Self)) {
```

Comment:
`Self` twice

Reply:
Oh, one of them should’ve been lowercase `self`.